### ISSUE TITLE
training yolo val mAP low

### ISSUE THREAD

**Issue by HaydenFaulkner** [dmlc/gluon-cv#840]
Training YOLO using the `train_yolov3.py` script with default arguments doesn't perform and match the log found in [the docs](https://raw.githubusercontent.com/dmlc/web-data/master/gluoncv/logs/detection/yolo3_darknet53_voc.log).

Python: 3.6.8
Mxnet: 1.4.1
Gluon: 0.4.0
Cuda: 10.0

The args are as follows:
```
Namespace(batch_size=64, data_shape=416, dataset='voc', epochs=200, gpus='0,1',
label_smooth=False, log_interval=100, lr=0.001, lr_decay=0.1, lr_decay_epoch='160,180',
 lr_decay_period=0, lr_mode='step', mixup=False, momentum=0.9, network='darknet53',
 no_mixup_epochs=20, no_random_shape=True, no_wd=False, num_samples=16551, 
num_workers=4, resume='', save_interval=10, save_prefix='yolo3_darknet53_voc', seed=233, 
start_epoch=0, syncbn=False, val_interval=1, warmup_epochs=0, warmup_lr=0.0, wd=0.0005)
```

The only two differences I can tell are `no_random_shape=True` and `syncbn=False`, however I am re-training with these flipped and similarly ~ 0 mAP after 3 epochs.

My best mAP log producing (after 70 epochs):
```
0000:	0.0000
0004:	0.0008
0005:	0.0023
0009:	0.0038
0010:	0.0069
0011:	0.0085
0013:	0.0096
0014:	0.0179
0016:	0.0202
0018:	0.0248
0022:	0.0309
0027:	0.0346
0030:	0.0405
0032:	0.0442
0034:	0.0507
0039:	0.0526
0040:	0.0536
0041:	0.0591
0045:	0.0625
0047:	0.0701
0050:	0.0724
0052:	0.0765
0053:	0.0807
0056:	0.0826
0057:	0.0930
0058:	0.0960
0061:	0.0969
0063:	0.0981
0064:	0.1033
0065:	0.1160
0067:	0.1199
```

Log output at epoch 70:
```
[Epoch 70][Batch 99], LR: 1.00E-03, Speed: 32.936 samples/sec, ObjLoss=22.087, BoxCenterLoss=6.444, BoxScaleLoss=2.084, ClassLoss=8.099
[Epoch 70][Batch 199], LR: 1.00E-03, Speed: 13.276 samples/sec, ObjLoss=22.044, BoxCenterLoss=6.444, BoxScaleLoss=2.083, ClassLoss=8.095
[Epoch 70] Training cost: 755.727, ObjLoss=22.017, BoxCenterLoss=6.443, BoxScaleLoss=2.082, ClassLoss=8.092
[Epoch 70] Validation: 
aeroplane=0.17818602073069206
bicycle=0.11919733680755705
bird=0.042505611946848855
boat=0.014124668435013263
bottle=0.09090909090909091
bus=0.07497190625855098
car=0.21510067511259742
cat=0.154457960909044
chair=0.009810333551340745
cow=0.08351810790835182
diningtable=0.0706671315310882
dog=0.1462674541939884
horse=0.08589083166915806
motorbike=0.16784611027660581
person=0.16790205233974093
pottedplant=0.002408187838651415
sheep=0.09687425435456931
sofa=0.10670158129460999
train=0.14739169785133233
tvmonitor=0.12951354474868979
mAP=0.10521222793337606
```

Running the SSD training script on same VOC data directory trains as expected, so the problem seems to be YOLO related. The SSD best mAP (after 10):
```
0000:	0.1368
0001:	0.3002
0002:	0.3828
0003:	0.4513
0004:	0.4881
0005:	0.5430
0007:	0.5653
0008:	0.6025
0009:	0.6123
0010:	0.6343
```

Any thoughts would be greatly appreciated,
Thanks :)

**Comment by HaydenFaulkner** [comment]
With further experimentation it appears the default learning rate of `0.001` is too high at the start to permit efficient learning. I found using a warmup learning rate of `0.0001` for the first 3 epochs solved the issue, generating the following mAP scores:
```
0000:	0.0372
0001:	0.1221
0002:	0.2112
0003:	0.3906
0004:	0.4416
0005:	0.5129
0006:	0.5717
0007:	0.5782
0009:	0.6473
0010:	0.6167
```

With settings `--warmup-lr 0.0001 --warmup-epochs 3`:
```
batch_size=64, data_shape=416, dataset='voc', epochs=200, gpus='0,1,2,3', label_smooth=False, 
log_interval=100, lr=0.001, lr_decay=0.1, lr_decay_epoch='160,180', lr_decay_period=0, lr_mode='step', 
mixup=False, momentum=0.9, network='darknet53', no_mixup_epochs=20, no_random_shape=False, 
no_wd=False, num_samples=16551, num_workers=16, resume='', save_interval=10, 
save_prefix='0006yolo3_darknet53_voc', seed=233, start_epoch=0, syncbn=True, val_interval=1, 
warmup_epochs=3, warmup_lr=0.0001, wd=0.0005
```
